apps/api/src/domain/games/memory-match/queries/memory-match-queries.fetch-by-id.e2e.spec.ts

```diff
@@ -151,4 +173,70 @@ describe(`when querying for a memory match round: fetch by Id`, () => {
             });
         });
     });
+
+    describe(`when the user is authenticated as a viewer`, () => {
+        beforeAll(async () => {
+            await setItUp(new CoscradUserWithGroups(nonAdminUser, []));
+        });
+
+        describe(`when there is a memory match round with the given ID`, () => {
+            describe(`when the round is public`, () => {
+                it(`should return the expected response`, async () => {
+                    const res = await request(app.getHttpServer()).get(
+                        buildDetailEndpoint(roundId)
+                    );
+
+                    expect(res.status).toBe(HttpStatusCode.ok);
+
+                    expect(res.body).toMatchSnapshot();
+                });
+            });
+
+            describe(`when the memory match round is private`, () => {
+                it(`should return not found`, async () => {
+                    const res = await request(app.getHttpServer()).get(
+                        buildDetailEndpoint(unpublishedRound.id)
+                    );
+
+                    expect(res.status).toBe(HttpStatusCode.notFound);
+                });
+            });
+        });
+    });
+
+    describe(`when the user is authenticated as a coscrad admin`, () => {
+        beforeAll(async () => {
+            await setItUp(new CoscradUserWithGroups(coscradAdmin, []));
+        });
+
+        describe(`when the memory match round is private`, () => {
+            it(`should return the expected result`, async () => {
+                const res = await request(app.getHttpServer()).get(
+                    buildDetailEndpoint(unpublishedRound.id)
+                );
+
+                expect(res.status).toBe(HttpStatusCode.ok);
+
+                expect(res.body).toMatchSnapshot();
+            });
+        });
+    });
+
+    describe(`when the user is authenticated as a project admin`, () => {
+        beforeAll(async () => {
+            await setItUp(new CoscradUserWithGroups(projectAdmin, []));
+        });
+
+        describe(`when the memory match round is private`, () => {
+            it(`should return the expected result`, async () => {
+                const res = await request(app.getHttpServer()).get(
+                    buildDetailEndpoint(unpublishedRound.id)
+                );
+
+                expect(res.status).toBe(HttpStatusCode.ok);
+
+                expect(res.body).toMatchSnapshot();
```

Comment:
This should be the only snapshot in the entire test. The point of the snapshot is to notify us of regressions that break the contract with the client. We pick the most comprehensive, most privileged query and snapshot the response body of this one (and only one) for a given endpoint. This avoids bloat of snapshots.